components/inputs/input-number.js

```diff
@@ -333,12 +340,17 @@ class InputNumber extends FocusMixin(LabelledMixin(SkeletonMixin(FormElementMixi
 				this._updateFormattedValue();
 			}
 		});
+
+		if (this.hiddenLabel && this.label && this.labelHidden) {
+			console.warn('d2l-input-number: Define the label with only label and label-hidden');
```

Comment:
In this scenario, we would want the consumer to just use `this.label` and `this.labelHidden` (and `input-text` would also ignore the `hiddenLabel`). Could move this warning into `input-text` as well.